src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentMetadataAsSourceFileProvider.cs

```diff
@@ -152,7 +191,7 @@ internal sealed class PdbSourceDocumentMetadataAsSourceFileProvider : IMetadataA
             var encoding = defaultEncoding ?? Encoding.UTF8;
             var sourceFileInfoTasks = sourceDocuments.Select(sd => _pdbSourceDocumentLoaderService.LoadSourceDocumentAsync(tempFilePath, sd, encoding, telemetry, useExtendedTimeout, cancellationToken)).ToArray();
             var sourceFileInfos = await Task.WhenAll(sourceFileInfoTasks).ConfigureAwait(false);
-            if (sourceFileInfos is null || sourceFileInfos.Where(t => t is null).Any())
+            if (sourceFileInfos is not [not null, ..])
```

Comment:
>not [not null, ..] [](http://example.com/codeflow?start=35&length=18)

Not sure what this condition means now. "does not start with non-null value"?

Reply:
lol yeah, I even tweeted about this one because I wasn't sure it was a good idea. Forgot it was in here too :)

It means "list is not null, has at least one element, and the first element is not null". I think old school is probably better for this one :)

Reply:
Cryptic.  Do we not want to check for any null elements?